silence/silence.go

```diff
@@ -276,6 +325,11 @@ func (s *Silences) setSilence(sil *pb.Silence) error {
 	if err != nil {
 		return err
 	}
+
+	_, err = s.mc.Add(sil)
+	if err != nil {
+		return err
+	}
```

Comment:
`setSilence()` still seems inconsistent. Either we do it for every set silence or none. In an HA setup with N peers, generally you have to expect about 1/N silences being created in the instance itself. The rest is fed via the mesh network.
So either set the cache on `Merge()` and `mergeDelta()` as described in my first comment or nowhere.

Doing it in `loadSnapshot()` sounds good so the first notifications are not unnecessarily delayed potentially. When new silences are set, that's relatively speaking very rare and not a performance concern.